packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -4,22 +4,39 @@
  */
 import { __ } from '@wordpress/i18n';
 import { Button } from '@wordpress/components';
-import { ENTER } from '@wordpress/keycodes';
+import { LEFT,
+	RIGHT,
+	UP,
+	DOWN,
+	BACKSPACE,
+	ENTER,
+} from '@wordpress/keycodes';
 
 /**
  * Internal dependencies
  */
 import { URLInput } from '../';
 
+const handleLinkControlOnKeyDown = ( event ) => {
```

Comment:
I remember @retrofox doing a lot of work on this keyboard press logic.

I assume we're re-adding this here because it's better to centralise the logic required to fix keyboard press events in the component itself rather than requiring the consuming component to handle it?

I'm only nervous because I know there were a number of edge cases that @retrofox was looking into which necessitated some of his work here.

Reply:
This is the exact behavior that was present before and yes it's just moved here  to centralize. You can see in  the "Buttons block" PR that it was duplicated.